### PR TITLE
add feishu plugin to SAA

### PR DESCRIPTION
**Pull request by wblu214**
add feishu plugin to SAA

### CODE REVIEW

**Comment by wblu214** [comment]
image
maven bulid error！！

**Comment by yuluo-yx** [comment]
you can run `mvn spring-javaformat:apply`

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml [hunk 1]
```diff
@@ -0,0 +1,106 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.1.1</version>
```
Comment (yuluo-yx) [inline community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml:8]: be consistent with the boot version of spring ai alibaba

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	<artifactId>spring-ai-alibaba-starter-plugin-feishu</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>spring-ai-alibaba-starter-plugin-feishu</name>
+	<description>FeiShu Tool for Spring AI Alibaba</description>
+
+	<properties>
+		<java.version>17</java.version>
+	</properties>
+
+	<dependencyManagement>
```
Comment (yuluo-yx) [inline community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml:21]: According to the transitive relationship of mvn, after introducing the above parent, there is no need to introduce it again here.

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+			<artifactId>spring-ai-spring-boot-autoconfigure</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-configuration-processor</artifactId>
+			<optional>true</optional>
+		</dependency>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-annotations</artifactId>
```
Comment (yuluo-yx) [inline community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml:55]: We shouldn't need this dependency. maybe can delete it

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-webflux</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.jsoup</groupId>
+			<artifactId>jsoup</artifactId>
+			<version>1.18.1</version>
+		</dependency>
+		<dependency>
```
Comment (yuluo-yx) [inline community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml:66]: ditto

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/main/java/com/alibaba/cloud/ai/plugin/feishu/FeiShuService.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
+package com.alibaba.cloud.ai.plugin.feishu;
+
+import com.google.gson.JsonParser;
```
Comment (yuluo-yx) [inline community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/main/java/com/alibaba/cloud/ai/plugin/feishu/FeiShuService.java:3]: I see that you have used jackson. The global situation should be unified. There is no need for gson here. What do you think?

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/main/java/com/alibaba/cloud/ai/plugin/feishu/config/FeiShuPluginConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ * @author wudihaoke214
+ * @author <a href="mailto:2897718178@qq.com">wudihaoke214</a>
+ */
+@Configuration
+@EnableConfigurationProperties(FeiShuProperties.class)
+public class FeiShuPluginConfiguration {
+    @Bean
+    @ConditionalOnMissingBean
+    @Description("Build FeiShu Client in Spring AI Alibaba")// description
+    @ConditionalOnProperty(
```
Comment (yuluo-yx) [inline community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/main/java/com/alibaba/cloud/ai/plugin/feishu/config/FeiShuPluginConfiguration.java:22]: As a plugin, I don’t think we need this configuration. cc @chickenlj

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/test/java/com/alibaba/cloud/ai/plugin/feishu/SpringAiAlibabaStarterPluginFeishuApplicationTests.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.alibaba.cloud.ai.plugin.feishu;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+/**
+ * @author wudihaoke214
+ * @author <a href="mailto:2897718178@qq.com">wudihaoke214</a>
+ */
+@SpringBootTest
+class SpringAiAlibabaStarterPluginFeiShuApplicationTests {
```
Comment (yuluo-yx) [inline community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/test/java/com/alibaba/cloud/ai/plugin/feishu/SpringAiAlibabaStarterPluginFeishuApplicationTests.java:10]: If there is no test class, you can delete it

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml (hunk 4 above)
Comment (wblu214) [reply community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml:66]: ok， but I saw that this dependency was used in the other plugins in the project

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/main/java/com/alibaba/cloud/ai/plugin/feishu/config/FeiShuPluginConfiguration.java (hunk 6 above)
Comment (wblu214) [reply community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/main/java/com/alibaba/cloud/ai/plugin/feishu/config/FeiShuPluginConfiguration.java:22]: In reader of Feishu,users need to configure their own appid and appSecret

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/test/java/com/alibaba/cloud/ai/plugin/feishu/SpringAiAlibabaStarterPluginFeishuApplicationTests.java (hunk 7 above)
Comment (wblu214) [reply community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/test/java/com/alibaba/cloud/ai/plugin/feishu/SpringAiAlibabaStarterPluginFeishuApplicationTests.java:10]: ok

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/main/java/com/alibaba/cloud/ai/plugin/feishu/FeiShuService.java (hunk 5 above)
Comment (wblu214) [reply community/plugins/spring-ai-alibaba-starter-plugin-feishu/src/main/java/com/alibaba/cloud/ai/plugin/feishu/FeiShuService.java:3]: ok

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml (hunk 3 above)
Comment (wblu214) [reply community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml:55]: ok

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml (hunk 2 above)
Comment (wblu214) [reply community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml:21]: ok

File: community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml (hunk 1 above)
Comment (wblu214) [reply community/plugins/spring-ai-alibaba-starter-plugin-feishu/pom.xml:8]: ok,I will update this groupid

**Pull request merged**